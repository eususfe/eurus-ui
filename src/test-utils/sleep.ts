/*
 * @Author: jiangxd
 * @Date: 2022-09-23 14:06:55
 * @LastEditTime: 2022-09-23 14:07:04
 * @LastEditors: jiangxd
 * @Description:
 * @FilePath: /eurus-ui/src/test-utils/sleep.ts
 */
const sleep = (time = 0) => new Promise(resolve => setTimeout(resolve, time));

export default sleep;
