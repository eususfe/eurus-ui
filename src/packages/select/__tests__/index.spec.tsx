import { mount } from '@vue/test-utils';
import { nextTick, ref } from 'vue';
import ESelect from '..';
import EOption from '../src/option';
import SelectDown from '@/packages/select-down';
import {expect} from "vitest";

const slots = {
  default: [
    <EOption value="1" label="基础用法1"/>,
    <EOption value="2" label="基础用法2"/>,
    <EOption value="3" label="基础用法3"/>,
  ]
};

describe('ESelect', () => {

  const modelValue = ref('1');

  it('should work with `modelValue` props', async () => {
    const wrapper = mount(ESelect, {
      props: {
        modelValue,
      },
      slots,
    });
    await nextTick();
    expect(wrapper.findComponent(SelectDown).find('.eu-select-down-control-single').text()).toMatchInlineSnapshot('"基础用法1"');
    expect(wrapper.findComponent(SelectDown).find('.eu-select-option-selected').text()).toMatchInlineSnapshot('"基础用法1"');

    modelValue.value = '2';
    await nextTick();
    expect(wrapper.findComponent(SelectDown).find('.eu-select-down-control-single').text()).toMatchInlineSnapshot('"基础用法2"');
    expect(wrapper.findComponent(SelectDown).find('.eu-select-option-selected').text()).toMatchInlineSnapshot('"基础用法2"');
  });

  it('should work with ', function () {

  });
});
