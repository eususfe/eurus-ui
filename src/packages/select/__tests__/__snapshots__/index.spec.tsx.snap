// Vitest Snapshot v1

exports[`ESelect > should work with \`modelValue\` props 1`] = `
"<div class=\\"eu-select-down eu-select\\" options=\\"\\">
  <div class=\\"eu-select-down-control eu-select-down-control-md\\" role=\\"listbox\\" tabindex=\\"0\\">
    <div class=\\"eu-select-down-control-single\\">基础用法1</div><input class=\\"eu-select-down-control-input\\">
    <!----><span class=\\"eu-icon down-icon\\" role=\\"link\\" tabindex=\\"0\\" style=\\"font-size: 20px;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\" class=\\"eu-icon eu-icon-chevron-down\\"><path fill=\\"currentColor\\" d=\\"M16 22 6 12l1.4-1.4 8.6 8.6 8.6-8.6L26 12z\\"></path></svg></span>
  </div>
  <transition-stub name=\\"slide-toggle\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
    <div class=\\"eu-select-down-pane\\" style=\\"display: none;\\" role=\\"listbox\\" tabindex=\\"0\\">
      <div class=\\"eu-select-down-pane-wrapper\\">
        <ul class=\\"scroll-pane\\">
          <li class=\\"eu-select-option eu-select-option-selected\\" role=\\"option\\" aria-selected=\\"true\\">基础用法1</li>
          <li class=\\"eu-select-option\\" role=\\"option\\" aria-selected=\\"false\\">基础用法2</li>
          <li class=\\"eu-select-option\\" role=\\"option\\" aria-selected=\\"false\\">基础用法3</li>
        </ul><span class=\\"down-arrow\\"></span>
      </div>
    </div>
  </transition-stub>
</div>"
`;

exports[`ESelect > should work with \`modelValue\` props 4`] = `
"<div class=\\"eu-select-down eu-select\\" options=\\"\\">
  <div class=\\"eu-select-down-control eu-select-down-control-md\\" role=\\"listbox\\" tabindex=\\"0\\">
    <div class=\\"eu-select-down-control-single\\">基础用法2</div><input class=\\"eu-select-down-control-input\\">
    <!----><span class=\\"eu-icon down-icon\\" role=\\"link\\" tabindex=\\"0\\" style=\\"font-size: 20px;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\" class=\\"eu-icon eu-icon-chevron-down\\"><path fill=\\"currentColor\\" d=\\"M16 22 6 12l1.4-1.4 8.6 8.6 8.6-8.6L26 12z\\"></path></svg></span>
  </div>
  <transition-stub name=\\"slide-toggle\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
    <div class=\\"eu-select-down-pane\\" style=\\"display: none;\\" role=\\"listbox\\" tabindex=\\"0\\">
      <div class=\\"eu-select-down-pane-wrapper\\">
        <ul class=\\"scroll-pane\\">
          <li class=\\"eu-select-option\\" role=\\"option\\" aria-selected=\\"false\\">基础用法1</li>
          <li class=\\"eu-select-option eu-select-option-selected\\" role=\\"option\\" aria-selected=\\"true\\">基础用法2</li>
          <li class=\\"eu-select-option\\" role=\\"option\\" aria-selected=\\"false\\">基础用法3</li>
        </ul><span class=\\"down-arrow\\"></span>
      </div>
    </div>
  </transition-stub>
</div>"
`;
